src/main/java/com/xpanxion/java/springboot/da1/demo/service/student9/MemberHistoryService9.java

```diff
@@ -73,35 +85,27 @@ public StringBuilder getMemberWorkOutHistory(int id){
        return newList;
     }
 
-    public StringBuilder getMemberWorkOutLength(int id, String type){
-        var stringBuilder = new StringBuilder();
+    public WorkoutLength9 getMemberWorkOutLength(int id, String type){
         if(type.equalsIgnoreCase("min")){
             var tempList = memberHistoryRepository9.getWorkoutLengthMin(id);
-            for(var temp : tempList){
-                var duration = Duration.between(temp.getCheckIn(), temp.getCheckOut());
-                stringBuilder.append("{");
-                stringBuilder.append("\n");
-                stringBuilder.append("LengthInMinutes: ").append(duration.toMinutes());
-                stringBuilder.append("\n");
-                stringBuilder.append("Member Id: ").append(temp.getMemberId().getMemberId());
-                stringBuilder.append("\n");
-                stringBuilder.append("date: ").append(temp.getCheckIn().toLocalDate());
-            }
+            var l = tempList.stream().map(m -> {
+                var duration = Duration.between(m.getCheckIn(), m.getCheckOut());
+                var memberId=  m.getMemberId().getMemberId();
+                var date = m.getCheckIn().toLocalDate();
+                return new WorkoutLength9(duration.toMinutes(), memberId, date);
```

Comment:
`retval = new WorkoutLength9(duration.toMinutes(), memberId, date);`

Reply:
return new WorkoutLength9(duration.toMinutes(), memberId, date);
            }).toList();
            
            retrieval =  l.get(0);

would something like this work?